src/main/java/com/clearblade/java/api/MqttClient.java

```diff
@@ -208,17 +208,28 @@ public void publish(String topic, String message) throws ClearBladeException {
 	 * @param qos quality of service
 	 * @throws ClearBladeException if publish fails
 	 */
-	public void publish(String topic, byte[] payload, int qos) throws ClearBladeException {
+	public void publish(String topic, byte[] payload, int qos, boolean retained) throws ClearBladeException {
 
 		try {
-			mqttClient.publish(topic, payload, qos, false);
+			mqttClient.publish(topic, payload, qos, retained);
 
 		} catch (MqttException e) {
 			String errmsg = String.format("(MqttClient) publish error: %s", e.getMessage());
 			throw new ClearBladeException(errmsg, e);
 		}
 	}
 
+	/**
+	 * Publishes the given message to the given topic.
+	 * @param topic topic to publish to
+	 * @param message message to publish
+	 * @param retained flag to set whether topic message is retained
+	 * @throws ClearBladeException if publish fails
+	 */
+	public void publish(String topic, String message, boolean retained) throws ClearBladeException {
+		publish(topic, message.getBytes(), defaultQualityOfService, retained);
+	}
+
```

Comment:
Can we move this function so that it appears before our final publish in line 211?

Reply:
Done. Re-committed, but also committed change to UserAuthExample.java. See my extended description. I needed the "sleep" at the end to make sure a retained message was deleted (if I uncommented the line that deletes the retained message).